File: backend/application/config/exceptions.py
"""Configuration Runtime Exception Hierarchy (Phase 14.3.1).

Defines custom exceptions for configuration runtime initialization, validation,
provider coordination, profile management, and source loading errors.
"""


class ConfigurationException(Exception):
    """Base exception for all Configuration Runtime errors."""

    pass


class ConfigurationInitializationError(ConfigurationException):
    """Raised when configuration runtime initialization fails."""

    pass


class ConfigurationValidationError(ConfigurationException):
    """Raised when configuration validation fails."""

    pass


class ConfigurationProviderError(ConfigurationException):
    """Raised when configuration provider encounters a runtime error."""

    pass


class ConfigurationProfileError(ConfigurationException):
    """Raised when configuration profile operations or activations fail."""

    pass


class ConfigurationSourceError(ConfigurationException):
    """Raised when loading or parsing a configuration source fails."""

    pass

File: frontend/src/runtime/commands/exceptions.ts
/**
 * Command Runtime Exception Hierarchy (Phase 16.6.1).
 *
 * Defines custom exceptions for command runtime initialization, provider errors,
 * execution failures, and command validation errors.
 */

export class CommandRuntimeException extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'CommandRuntimeException';
    Object.setPrototypeOf(this, new.target.prototype);
  }
}

export class CommandInitializationException extends CommandRuntimeException {
  constructor(message: string) {
    super(message);
    this.name = 'CommandInitializationException';
    Object.setPrototypeOf(this, new.target.prototype);
  }
}

export class CommandProviderException extends CommandRuntimeException {
  constructor(message: string) {
    super(message);
    this.name = 'CommandProviderException';
    Object.setPrototypeOf(this, new.target.prototype);
  }
}

export class CommandExecutionException extends CommandRuntimeException {
  constructor(message: string) {
    super(message);
    this.name = 'CommandExecutionException';
    Object.setPrototypeOf(this, new.target.prototype);
  }
}

export class CommandValidationException extends CommandRuntimeException {
  constructor(message: string) {
    super(message);
    this.name = 'CommandValidationException';
    Object.setPrototypeOf(this, new.target.prototype);
  }
}
